### PR TITLE
feat: added maxHeight and vertical scroll in chip Input

### PR DESCRIPTION
**Pull request by geeky-abhishek**

### CODE REVIEW

File: src/components/inputs/ChipInput.tsx [hunk 1]
```diff
@@ -717,6 +730,18 @@ const ChipInput: ChipInput = React.forwardRef<HTMLDivElement, ChipInputProps>(fu
...
 	);
 
+	useEffect(() => {
+		const newHeight = hScrollContainerRef?.current?.offsetHeight || 24;
+		setContainerHeight(newHeight);
+	}, [items]);
+
+	const isOverflowing = useMemo(() => {
+		if (/px$/.test(maxHeight)) {
+			return containerHeight >= Number(maxHeight?.split('px')[0]) - 15;
```
Comment (beawar) [inline src/components/inputs/ChipInput.tsx]: I don't like this check very much. What if I want to set vh/em/% as unit measure for the component max height?

File: src/components/inputs/ChipInput.tsx [hunk 2]
```diff
@@ -69,19 +74,23 @@ const HorizontalScrollContainer = styled.div<{ wrap: 'nowrap' | 'wrap'; hasLabel
...
+	max-height: ${({ maxHeight }): string => maxHeight};
+	overflow-y: scroll;
+	${({ hasLabel, wrap, theme, isOverflowing }): SimpleInterpolation =>
 		hasLabel &&
 		wrap === 'wrap' &&
 		css`
 			&::before {
 				content: '';
-				min-height: calc(${theme.sizes.font.extrasmall} * 1.5);
+				min-height: (${isOverflowing ? 'fit' : `calc(${theme.sizes.font.extrasmall} * 1.5)`});
```
Comment (beawar) [inline src/components/inputs/ChipInput.tsx]: This is not working: external brackets are preventing the rule to be applied on browser

**Review by beawar — COMMENTED** [review]
I'm not a fan of this solution, next days I will think about an alternative. Also, I think we could ask UX to uniform the scrollbar to be visible for both horizontal or vertical scroll, in order to improve usability and reduce complexity of the code required to style the component.

File: src/components/inputs/ChipInput.tsx (hunk 1 above)
Comment (geeky-abhishek) [reply src/components/inputs/ChipInput.tsx]: @beawar I updated the condition which will work for all possible sizes ,thanks for pointing it out

File: src/components/inputs/ChipInput.tsx (hunk 2 above)
Comment (geeky-abhishek) [reply src/components/inputs/ChipInput.tsx]: removed

**Comment by geeky-abhishek** [comment]
> I'm not a fan of this solution, next days I will think about an alternative. Also, I think we could ask UX to uniform the scrollbar to be visible for both horizontal or vertical scroll, in order to improve usability and reduce complexity of the code required to style the component.

I'll ask @ Alice for the feedback

**Comment by beawar** [comment]
I've seen last changes, thank you for the time you spent to improve the solution. I think we are almost there. There is still a case not well covered: there is a lot of space between label and chips when vertical scroll is active. To solve this, why don't we change entirely the way space is reserved for the label, using the margin-top as you did instead of the :before?
It seems to me that doing so we are able to remove also the logic to check if div is overflowing, simplifying a little bit the component.

Check this patch to see if it fits (to apply the patch, unzip and then: ```git apply chipInput-vertical-scroll.patch```
[chipInput-vertical-scroll.zip](https://github.com/Zextras/carbonio-design-system/files/8926098/chipInput-vertical-scroll.zip)

PS: I've also revomed references to horizonal scroll in component and variable names since we are dealing with vertical scroll too now.

PPS: align this branch to [CDS-24-design-system-demo-review](https://github.com/Zextras/carbonio-design-system/tree/CDS-24-design-system-demo-review) before applying the patch

**Comment by beawar** [comment]
I've seen last changes, thank you for the time you spent to improve the solution. I think we are almost there. There is still a case not well covered: there is a lot of space between label and chips when vertical scroll is active. To solve this, why don't we change entirely the way space is reserved for the label, using the margin-top as you did instead of the :before?
It seems to me that doing so we are able to remove also the logic to check if div is overflowing, simplifying a little bit the component.

Check this patch to see if it fits (to apply the patch, unzip and then: ```git apply chipInput-vertical-scroll.patch```
[chipInput-vertical-scroll.zip](https://github.com/Zextras/carbonio-design-system/files/8926098/chipInput-vertical-scroll.zip)

PS: I've also revomed references to horizonal scroll in component and variable names since we are dealing with vertical scroll too now.

PPS: align this branch to [CDS-24-design-system-demo-review](https://github.com/Zextras/carbonio-design-system/tree/CDS-24-design-system-demo-review) before applying the patch

**Comment by beawar** [comment]
@geeky-abhishek ping

**Comment by geeky-abhishek** [comment]
> 

hi @beawar sorry I was on vacation and have resume yesterday only.I'll implement the fix this afternoon

**Review by beawar — APPROVED** [review]
LGTM

**Pull request merged**